bfasst/flows/gather_impl_data_flow.py

```diff
@@ -0,0 +1,135 @@
+"""GatherImplData flow"""
```

Comment:
Can you expand this docstring?  I'm also a bit confused on the naming of this.  What does "gather implementation" mean?